File: hackerrank/day18/main.js
class Solution {
    constructor() {
        this.stack = [];
        this.queue = [];
    }

    pushCharacter(ch) {
        this.stack.push(ch);
    }

    enqueueCharacter(ch) {
        this.queue.push(ch);
    }

    popCharacter() {
        return this.stack.pop();
    }

    dequeueCharacter() {
        return this.queue.shift();
    }
}

File: hackerrank/day18/main.cs
using System.Collections.Generic;

public class Solution
{
    private Stack<char> stack = new Stack<char>();
    private Queue<char> queue = new Queue<char>();

    public void pushCharacter(char ch)
    {
        stack.Push(ch);
    }

    public void enqueueCharacter(char ch)
    {
        queue.Enqueue(ch);
    }

    public char popCharacter()
    {
        return stack.Pop();
    }

    public char dequeueCharacter()
    {
        return queue.Dequeue();
    }
}